src/lib.rs

```diff
@@ -63,20 +63,71 @@ use serde_json::Value;
 use std::{env, io, io::Read, iter, path::Path, time::Duration};
 use url::form_urlencoded;
 
+macro_rules! API_REFERENCE_URL {
+    () => {
+        "https://docs.docker.com/engine/api"
+    };
+}
+
+macro_rules! API_VERSION {
+    () => {
+        "v1.41"
+    };
+}
+
+macro_rules! _api_ep {
+    (start) => {
+        concat!(
+            "[Api Reference](",
+            API_REFERENCE_URL!(),
+            "/",
+            API_VERSION!(),
+        )
+    };
+    (index) => {
+        concat!(_api_ep! {start}, ")")
+    };
+    ($category:expr, $ep:expr) => {
+        concat!(_api_ep! {start}, "/#", $category, "/", $ep, ")",)
+    };
+}
+
+macro_rules! _api_doc {
+    ($api:expr, $($extra:expr),*, @: /* this is used to disambiguate expr from tt */ $($tt:tt)*) => {
+        $(
+            #[doc = $extra]
+            #[doc = "\n"]
+        )*
+        #[doc = $api]
+        $($tt)*
+    }
+}
+
+macro_rules! api_doc {
+    ($category:expr, $ep:expr, $($comments:expr),*, => $($tt:tt)*) => {
+        _api_doc! {_api_ep!($category, $ep), $($comments),*, @: $($tt)*}
+    };
+    ($($comments:expr),*, => $($tt:tt)*) => {
+        _api_doc! {_api_ep!(index), $($comments),*, @: $($tt)*}
+    };
+}
+
 /// Represents the result of all docker operations
 pub type Result<T> = std::result::Result<T, Error>;
 
-/// Entrypoint interface for communicating with docker daemon
+api_doc! {
+"Entrypoint interface for communicating with docker daemon", =>
 #[derive(Clone)]
 pub struct Docker {
     transport: Transport,
-}
+}}
```

Comment:
Hmmm... Now that I'm looking at the code and usage of `api_doc!()` I'm less of a fan. It's a lot of magic and not very readable (it even breaks rustfmt) for a pretty small gain.

Reply:
Yeah, after second thought I think you're right that it's not worth the added complexity, although having a direct link to API in docs is actually a nive quality of life improvement. 

I went ahead and created #259 for documenting api endpoints, maybe we can come up with another way of solving this.